rfcs/005-group-stats.md

```diff
@@ -0,0 +1,117 @@
+|           | Student 2   |      X       | ... |      Y      |
+|           | Student 3   |      Y       | ... |      Y      |
+|           | Student 4   |      Y       | ... |      Y      |
+
+(In this example, Student 2 did not work with their assigned group on assessment 1.)
+
+*Role Participation*
+
+Filters: [Include student data from any group]
+
+| Student ID | Role 1 | Role 2 | Role 3 | Role 4 |
+|------------|--------|--------|--------|--------|
+| Student 1  |    3   |   1    |   1    |   1    |
+| Student 2  |    1   |   1    |   1    |   1    |
+| Student 3  |    2   |   1    |   2    |   1    |
+| Student 4  |    0   |   2    |   1    |   2    |
+
+*Submission Count* 
+
+Filters: [Include student data from assigned group only]
+
+| Student ID | Assessment 1 | ... | Assessment N | Total | Percentage of Group |
+|------------|--------------|-----|--------------|-------|---------------------|
+| Student 1  |      5       | ... |      0       |  30   |        34.88        |
+| Student 2  |      1       | ... |      2       |   8   |         9.30        |
+| Student 3  |      8       | ... |     12       |  28   |        32.55        |
+| Student 4  |      2       | ... |      7       |  20   |        23.26        |
+
+
+# Motivation
+As group assessments are now a familiar feature of PrairieLearn, is important that instructors are provided support to gain insight into their class's behaviors during group assessments, specifically a better understanding at the group-level. Concerns of students working with their assigned group, free-loading and domination, and adhering to class policies (e.g. group assessments with roles, students must take on each role at least once) still remain and left up to the instructor to address either relying on students to comply without guidance or enforcement or having pull data as needed and creating scripts or manually determining how students are doing, perhaps when the opportune time to intervene has since passed.
+
+## Additional Motivation Context for Design Choices
+As PrairieLearn stands right now, instructors can either pre-assign groups, where a student can get credit without joining, or instructors can allow students to create and join their own groups, which give the ability to join any group or depending on configuration, create a group of one. Providing group insights can help inform instructors quickly of group performances.
+
+## Goals: 
+* Deliver Group Statistics so instructors have a high-level idea of groups' performances; quickly identify potentially struggling groups, signs of potential free-loading or domination
+* Allow filters and parameters to serve a variety of instructor needs
+
+# Proposed Solution
+Adding a Group Statistics tab (`pages/instructorGroupStatistics`) to the Course Instance-level where PrairieLearn will provide insights on group performances across group submissions. 
+
+**Input:** CSV of desired group assignments to compare against. 
```

Comment:
Will these uploaded CSVs be persisted and associated with the course so that folks can reuse them? Or will instructors be expected to re-upload them every time they want to compute new statistics?

Reply:
This I wasn't as sure about. While re-uploading every time might be annoying and it would be nice to be able to re-use, it feels like quite a bit of modification to have that reuse that I wasn't sure was desirable, but that could also be my lack of expertise.

Reply:
Yeah, this would require new database tables, a fair amount of new UI, and so on. In the interest of shipping something that's immediately useful to folks, I'd be inclined to say that we should require instructors to upload a new file every time.

Reply:
Fully makes sense and I agree, uploading a file every time will be the way to go then, thanks!